doc/image_proc.rst

```diff
@@ -0,0 +1,351 @@
+.. include:: include/links.rst
+
+.. _image_proc:
+
+======================
+Basic Image Processing
+======================
+
+Here we describe the basic image processing steps performed by ``PypeIt``,
+specifically their order, salient details, and how to toggle specific steps
+using the :ref:`pypeit_file`.  This description is meant to be general to *all*
+spectrographs.  For instrument-specific advice, see :ref:`spec_details`.
+
+Unless otherwise stated, the relevant parameters governing basic image
+processing are kept by :class:`~pypeit.par.pypeitpar.ProcessImagesPar`; see
+:ref:`pypeitpar`.  Importantly, however, changing the default parameters for
+some frame types will lead to faults; see :ref:`proc_algorithm` and
+:ref:`workflow`.
+
+.. _overview:
+
+Overview
+========
+
+We provide a very general approach to image processing with hooks to toggle each
+step, as appropriate for different frame types and/or different instruments.
+Generally, we treat pixel values, :math:`p`, in an observed frame as containing
+the following components:
+
+.. math::
+
+    p = O + B + (C + D\ t_{\rm exp}) / \gamma
+
+where:
+
+    - :math:`O` is a frame-dependent bias estimate (in ADU) using image overscan
+      regions,
+    - :math:`B` is a longer-term pixel-by-pixel bias estimate (in ADU) using
+      bias images,
+    - the quantity :math:`C=c/s` is the number of electron counts excited by
+      photons hitting the detector,
+    - :math:`1/s` is an efficiency factor (one of many) that accounts for relative
+      throughput factors (see below) that can be measured from flat-field frames,
+    - :math:`D` is the rate at which the detector generates thermal electrons (in e-/s),
+    - :math:`t_{\rm exp}` is the exposure time in seconds, and
+    - :math:`\gamma` is the amplifier gain in e- per ADU.
+
+By "relative throughput," we mean the aggregate of relative (to, say, the center
+of the field) telescope+instrument+detector efficiency factors that can be
+measured from flat-field images, like pixel-to-pixel differences in quantum
+efficiency, vignetting, and the wavelength-dependent grating efficiency.  The
```

Comment:
Wavelength dependent grating efficiency is not taken out as part of s. KCWI does perform a speculum correction,  but that is only a relative correction (to the average of the slits) I believe. Let's try to use language that is specific to pypeit, like illumflat, pixelflat etc. and not different language otherwise this will be confusing. Ideally we could link to the documentation on those master files here. I think you need @rcooke-ast to help you with the KCWI specific things. If you want to give me permission to edit directly am happy to do so.

Reply:
Thanks for asking, and, yes, please feel free to edit directly.  Thanks for pointing out the error concerning the wavelength dependence.  It would definitely be useful to include references to things like `pixelflat` and `illumflat`.  But I'd prefer we do this in a way that links pypeit jargon to the effects that they're trying to address.  At the moment, the text only includes that latter, and it's a good idea to draw the direct link to the former.

Reply:
I updated this. Please take a look.